### PR TITLE
MAINT: backports before 1.9.0rc3

### PR DESCRIPTION
**Pull request by tylerjereddy**
- Backport PRs:
  - gh-16514
  - gh-16549
  - gh-16551
  - gh-16563
  - gh-16429
  - gh-16572
  - gh-16581
  - gh-16586
  - gh-16593
  - gh-16599
  - gh-16605
  - gh-16622
  - gh-16638
  - gh-16664
- Update release notes to reflect above backports + other separate merges to the maintenance branch
- set the version to "1.9.0rc3 unreleased" (normally done via direct push)

@rgommers @h-vetinari anything else I'm missing before `1.9.0rc3` is ready to go? If you think we're about ready, I could try to get it out before the travel to SciPy 2022 conference starts.

### CODE REVIEW

**Comment by mdhaber** [comment]
@rgommers mentioned that gh-16418 needs to be resolved. gh-16429 would do that, but it hasn't been merged.

**Review by rgommers — COMMENTED** [review]
@tylerjereddy I don't think there's a point in doing an RC3 release with just these changes, because they're all safe changes so there's not much value. I would suggest to make RC3 the final release candidate, which means we want to have closed the five remaining items in https://github.com/scipy/scipy/issues?q=is%3Aopen+is%3Aissue+milestone%3A1.9.0.

Looking at those items:
- I think you can just close the file permissions and the doc build ones
- I will look at merging the `stats.mode` PR from @mdhaber today, and then you can include the backport in this PR
- I will deal with the MKL one today as well.
- That leaves the last issue, for which we need a new `meson-python` release. @FFY00 aims to do that by the end of next week.

**Review by h-vetinari — CHANGES_REQUESTED** [review]
Looks good from my side, but still needs a mention of the increased windows build requirements. Something like:
> Building SciPy on windows with MSVC now requires at least the vc142 toolset (available in Visual Studio 2019 and higher).

**Comment by tylerjereddy** [comment]
Ok, sounds good. So, no pressure to release in the next 48 hours which is good.

**Comment by h-vetinari** [comment]
I think https://github.com/scipy/scipy/pull/16622 would be worth another rc (about +-20k lines in HiGHS submodule), once it lands.

**Comment by tylerjereddy** [comment]
> I also just noticed that the warning box at the top of https://scipy.github.io/devdocs/dev/contributor/meson.html is pretty outdated

Although my experience trying to help folks build from source using `meson` on Windows at the SciPy sprint is still consistent with the warning box statement that Windows support is not super wide re: compiler toolchains.

**Comment by rgommers** [comment]
> I also just noticed that the warning box at the top of https://scipy.github.io/devdocs/dev/contributor/meson.html is pretty outdated by now and should probably still be updated for 1.9

Good point, I opened gh-16638 to address this.

**Comment by tylerjereddy** [comment]
`meson-python` `0.7.0` was released about an hour ago, so my plan is to update this backport PR with the backports that have accumulated over the past two weeks--probably on Friday (tomorrow).

**Comment by tupui** [comment]
We also need to keep in mind the theme update. I asked if a release is imminent and will let you know. E.g. if the release happen after the RC and you build the doc for the release, then we don't have a search bar because it needs an update on our side (#16660)

**Comment by FFY00** [comment]
@tylerjereddy let me know if you have any issues!

**Comment by rgommers** [comment]
> We also need to keep in mind the theme update. I asked if a release is imminent and will let you know. E.g. if the release happen after the RC and you build the doc for the release, then we don't have a search bar because it needs an update on our side (#16660).

That still seems blocked, but that should not be a problem for `1.9.0rc3` or the final release, because docs can easily be rebuilt afterwards.

> `meson-python` `0.7.0` was released about an hour ago, so my plan is to update this backport PR with the backports that have accumulated over the past two weeks--probably on Friday (tomorrow).

I'll try to merge the remaining open PRs with a `backport-candidate` label this morning, and will open a separate PR for updating dependency versions in 1.9.x

**Comment by rgommers** [comment]
@tylerjereddy everything should be good to go now. I removed the `backport-candidate` label from the PRs you already have included in this PR. There are 9 merged PRs left to add to this backport: https://github.com/scipy/scipy/issues?q=label%3Abackport-candidate+is%3Aclosed

**Comment by tylerjereddy** [comment]
@rgommers thanks so much

**Comment by tylerjereddy** [comment]
Oy, some crazy merge conflicts in `stats`, I'll probably ping for a review on those.

**Comment by mdhaber** [comment]
I can take a look.

**Comment by tylerjereddy** [comment]
I'll let you know once I've completed the cherry-picks--I've flagged the commits with conflicts where I did manual work so I'll post those here when I push in.

**Comment by tylerjereddy** [comment]
Added all the available backports and updated release notes accordingly.

@mdhaber I was quite concerned about `stats` merge conflicts related to these hashes:
- 92ee2ae (it looked like the example docstring data had changes vs. `main`)
- 66c93a7234e32bb6bc8551e3b7f4fe36e14ea5c6
- 16e2c6faa40409439be468ef8c758531a044c1ff

File: scipy/stats/_stats_py.py [hunk 1]
```diff
@@ -593,29 +605,50 @@ def mode(a, axis=0, nan_policy='propagate'):
     ...               [5, 3, 0, 5],
     ...               [4, 7, 5, 9]])
     >>> from scipy import stats
-    >>> stats.mode(a)
-    ModeResult(mode=array([3, 1, 0, 0]), count=array([1, 1, 1, 1]))
+    >>> stats.mode(a, keepdims=True)
+    ModeResult(mode=array([[3, 0, 6, 1]]), count=array([[4, 2, 2, 1]]))
```
Comment (tylerjereddy) [inline scipy/stats/_stats_py.py:609]: @mdhaber There's definitely something wrong here--when I was rebasing I was worried that the data in `a` was different vs. `main` I think, and that's related to the current `stats` refguide fail I believe.

Comment (mdhaber) [reply scipy/stats/_stats_py.py:609]: Yeah the input data changed in https://github.com/scipy/scipy/pull/16629. If we're not changing the input data, we don't want to change the output data. But due to https://github.com/scipy/scipy/pull/16429, the output should be be 2D.

```suggestion
    ModeResult(mode=array([[3, 1, 0, 0]]), count=array([[1, 1, 1, 1]]))
```
I think this is right.

File: scipy/stats/tests/test_distributions.py [hunk 2]
```diff
@@ -1017,62 +1027,52 @@ def _test_moments_one_range(self, a, b, expected, rtol=1e-7):
...
+         -0.282786110727154,
+         0.6161417353578293,
+         -0.5393018494027877,
+         -0.20582065135274694],
+        [-10, -9,
+         -9.108456288012409,
+         0.011448805821636248,
+         -1.8985607290949496,
+         5.0733461105025075],
+    ]
```
Comment (mdhaber) [inline scipy/stats/tests/test_distributions.py:1068]: This looks right. You mentioned https://github.com/scipy/scipy/commit/66c93a7234e32bb6bc8551e3b7f4fe36e14ea5c6, which shows some additional cases, but those were removed by the end of gh-16593, which we're backporting.

**Comment by mdhaber** [comment]
And re: https://github.com/scipy/scipy/commit/16e2c6faa40409439be468ef8c758531a044c1ff, I think that's all right. In this PR, `truncnorm` looks exactly like it does in `main`, which I think is correct because we're backporting all the recent changes to `truncnorm`.

**Comment by h-vetinari** [comment]
This comment from further up is still open.

> Looks good from my side, but still needs a mention of the increased windows build requirements. Something like:
> 
> > Building SciPy on windows with MSVC now requires at least the vc142 toolset (available in Visual Studio 2019 and higher).

**Comment by tylerjereddy** [comment]
I pushed in the suggested fixes for `stats.mode` and the MSVC windows build relnotes adjustment. The placement of the latter note is perhaps debatable, I think I put in `Other` section

**Review by rgommers — APPROVED** [review]
This looks good to go now. I updated the PR description with all backports. In it goes - looking forward to RC3:) Thanks @tylerjereddy & all

**Comment by tylerjereddy** [comment]
Wheel builds have started--I did notice an Azure warning that macOS 10.15 environment is deprecated, hopefully can ignore for now.

**Comment by h-vetinari** [comment]
> Wheel builds have started--I did notice an Azure warning that macOS 10.15 environment is deprecated, hopefully can ignore for now.

You might be interested in this [comment](https://github.com/conda-forge/conda-forge.github.io/issues/1798#issuecomment-1191377086) - basically, it should be possible to use the 10.15 SDK also in the macos-11 image, and consequently target builds as old as macos 10.6 (note that conda-forge brings its own SDK so the follow-up comments discuss how this doesn't apply to us there).

**Pull request merged**